sql/analyzer/triggers.go

```diff
@@ -357,3 +358,74 @@ func orderTriggersAndReverseAfter(triggers []*plan.CreateTrigger) []*plan.Create
 func triggerEventsMatch(event plan.TriggerEvent, event2 string) bool {
 	return strings.ToLower((string)(event)) == strings.ToLower(event2)
 }
+
+// wrapPlansWithTriggers wraps the entire tree iff it contains a trigger, allowing rollback when a trigger errors
+func wrapPlansWithTriggers(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel RuleSelector) (sql.Node, transform.TreeIdentity, error) {
```

Comment:
Call this wrapWritesWithRollback

We're going to end up doing this a lot of places, I bet